components/fonts/platform/freetype/ohos/font_list.rs

```diff
@@ -65,23 +103,265 @@ struct FontList {
     aliases: Vec<FontAlias>,
 }
 
+fn enumerate_font_files() -> io::Result<Vec<PathBuf>> {
+    let mut font_list = vec![];
+    for elem in fs::read_dir(OHOS_FONTS_DIR)? {
+        if let Ok(e) = elem {
+            if e.file_type().unwrap().is_file() {
+                let name = e.file_name();
+                let raw_name = name.as_bytes();
+                if raw_name.ends_with(b".ttf".as_ref()) || raw_name.ends_with(b".ttc".as_ref()) {
+                    debug!("Found font {}", e.file_name().to_str().unwrap());
+                    font_list.push(e.path())
+                }
+            }
+        }
+    }
+    Ok(font_list)
+}
+
+fn detect_hos_font_style(font_modifiers: &[&str]) -> Option<String> {
+    if font_modifiers.contains(&"Italic") {
+        Some("italic".to_string())
+    } else {
+        None
+    }
+}
+
+// Note: The weights here are taken from the `alias` section of the fontconfig.json
+fn detect_hos_font_weight_alias(font_modifiers: &[&str]) -> Option<i32> {
```

Comment:
This function infers the `weight` attribute for `HarmonyOS Sans` fonts, based on suffixes such as `Medium`, `Bold` etc. The numbers here are taking from the alias section of fontconfig.json which I added below.
I'm not sure if I should use this weight here, or if the alias weight, should only be added to the `FontAlias`es.
Happy for any recommendations.

```json
{
  "fontdir": ["fonts"],
  "generic": [
    {
      "family": "HarmonyOS Sans",
      "alias": [
        {
          "HarmonyOS-Sans": 0
        },
        {
          "HarmonyOS-Sans-Light": 100
        },
        {
          "HarmonyOS-Sans-Regular": 400
        },
        {
          "HarmonyOS-Sans-Medium": 700
        },
        {
          "HarmonyOS-Sans-Bold": 900
        }
      ],
      "adjust": [
        {
          "weight": 50, "to": 100
        },
        {
          "weight": 80, "to": 400
        },
        {
          "weight": 100, "to": 700
        },
        {
          "weight": 200, "to": 900
        }
      ]
    },
    {
      "family": "HarmonyOS Sans Condensed",
      "alias": [
        {
          "HarmonyOS-Sans-Condensed": 0
        }
      ]
    },

```